modules/core/04-channel/keeper/keeper.go

```diff
@@ -653,8 +671,33 @@ func (k Keeper) HasPruningSequenceStart(ctx sdk.Context, portID, channelID strin
 	return store.Has(host.PruningSequenceStartKey(portID, channelID))
 }
 
-// PruneAcknowledgements prunes packet acknowledgements from the store that have a sequence number less than or equal to the pruning sequence.
+// PruneAcknowledgements prunes packet acknowledgements from the store that have a sequence number less than or equal to the last packet send.
 // The number of packet acknowledgements pruned is equal to limit. Pruning only occurs after a channel has been upgraded.
-func (Keeper) PruneAcknowledgements(ctx sdk.Context, portID, channelID string, limit, pruningSequence uint64) error {
-	return nil
+//
+// The pruningSequence keeps track of the packet acknowledgement that can be pruned next. When the pruning sequence reaches
+// the last send sequence, pruning is complete.
+func (k Keeper) PruneAcknowledgements(ctx sdk.Context, portID, channelID string, limit, pruningSequenceStart uint64) (uint64, error) {
+	pruningSequenceEnd, found := k.GetPruningSequenceEnd(ctx, portID, channelID)
+	if !found {
+		return pruningSequenceStart, errorsmod.Wrapf(types.ErrPruningSequenceStartNotFound, "port ID (%s) channel ID (%s)", portID, channelID)
+	}
+
+	limit = pruningSequenceStart + limit
+	for ; pruningSequenceStart < limit; pruningSequenceStart++ {
+		// break if the pruning sequence reaches the last send sequence
+		if pruningSequenceStart > pruningSequenceEnd {
```

Comment:
```suggestion
		if pruningSequenceStart >= pruningSequenceEnd {
```
We don't want to prune the next sequence send ack https://github.com/cosmos/ibc-go/blob/04-channel-upgrades/modules/core/04-channel/keeper/upgrade.go#L586